### PR TITLE
REGD-107-test

### PR DESCRIPTION
**Pull request by mingjiecn**

### CODE REVIEW

File: cypress/fixtures/example.json
Comment (ottojolanki) [inline cypress/fixtures/example.json:1]: Is this file necessary?

File: cypress/support/commands.js
Comment (ottojolanki) [inline cypress/support/commands.js:1]: Does not seem to contain any code?

**Comment by ottojolanki** [comment]
Would probably be good to have someone who knows more JS than I also have a look. @forresttanaka could you also check this one?

File: cypress/fixtures/example.json
Comment (mingjiecn) [reply cypress/fixtures/example.json:1]: No. I will delete them.

File: cypress/support/commands.js
Comment (mingjiecn) [reply cypress/support/commands.js:1]: Those are auto generated by cypress.

**Comment by mingjiecn** [comment]
> 

I slacked @forresttanaka for code review

File: pages/index.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
+import React from "react";
```
Comment (forresttanaka) [inline pages/index.js]: Is this needed? It doesn’t hurt anything, but I don’t think you need it without Typescript.

File: pages/summary.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  data: PropTypes.string.isRequired,
+};
+
+export async function getServerSideProps({ query }) {
+  const queryElements = [];
+  for (const [key, value] of Object.entries(query)) {
+    const element = key + "=" + value;
+    queryElements.push(element);
+  }
+  const queryString = queryElements.join("&");
```
Comment (forresttanaka) [inline pages/summary.js]: `getQueryStringFromServerQuery()` does this composition of a query string from the server query. The code here won’t work right if you have repeated query-string keys (e.g. "type=AnalysisSet&type=MeasurementSet") because the keys appear only once in `query` but its value is an array of values, which won’t work right here. This also applies to search.js, though this code isn’t in the diffs in that case.

**Review by forresttanaka — CHANGES_REQUESTED** [review]
This looks really good! My feedback isn’t about the tests, but your server query-string processing can get tricked by NextJS, and you already have a function that handles the tricky cases.

**Comment by forresttanaka** [comment]
Reviewing…done.

File: pages/search.js [hunk 3]
```diff
@@ -19,23 +18,14 @@ Page.propTypes = {
...
   const url = API_URL + "/search?" + queryString;
-  console.log(url);
-
   const response = await fetch(url);
   const data = await response.text();
   const breadcrumbs = [
     {
       title: "Search",
-      href: "query",
+      href: "/search?" + queryString,
```
Comment (forresttanaka) [inline pages/search.js]: One choice to make is whether to allow string concatenation through addition or not. Our ENCODE rules don’t allow this — you have to use string templates. I didn’t realize the igvf-ui one did allow this, so I’ll change that. But you might decide to allow it for RegulomeDB.

**Review by forresttanaka — APPROVED** [review]
😎

File: pages/search.js (hunk 3 above)
Comment (mingjiecn) [reply pages/search.js]: Sounds good. I will use string templates then, more readable.
